crates/rpc/src/methods/send_transaction.rs

```diff
@@ -179,22 +155,42 @@ impl SendTransaction {
             .ok_or(Error::WalletNameNotFound(self.wallet_name.clone()))?
             .clone();
 
-        let signer = wallet
-            .build_signer(self.network.chain_id, &self.wallet_path)
-            .await?;
-
         let url = self
             .network
             .http_url
             .parse()
             .map_err(|_| Error::CannotParseUrl(self.network.http_url.clone()))?;
 
-        let provider = ProviderBuilder::new()
-            .with_recommended_fillers()
-            .wallet(signer.to_wallet())
-            .on_http(url);
+        let is_dev_network = self.network.is_dev() && self.network.chain_id == 31337;
+
+        match wallet {
+            Wallet::Impersonator(ref wallet) => {
+                if is_dev_network {
+                    let account = wallet.get_address(&self.wallet_path).await?;
+                    let provider = ProviderBuilder::new()
+                        .with_recommended_fillers()
+                        .on_http(url);
+
+                    // TODO: maybe we can find a way to only do this once for every account,
+                    // or only call anvil_autoImpersonate once for the whole network,
+                    // instead of making this request for every single transaction.
+                    // this is just a minor optimization, though
+                    provider.anvil_impersonate_account(account).await?;
+                    self.provider = Some(Box::new(provider));
+                }
+            }
+            _ => {
+                let signer = wallet
+                    .build_signer(self.network.chain_id, &self.wallet_path)
+                    .await?;
+                let provider = ProviderBuilder::new()
+                    .with_recommended_fillers()
+                    .wallet(signer.to_wallet())
+                    .on_http(url);
+                self.provider = Some(Box::new(provider));
+            }
+        };
```

Comment:
just to confirm, will the default match case also match when the if in `if is_dev_network` case fails?

Reply:
not, but you can make it so:

```rust
match wallet {
  Wallet::Impersonator(ref wallet) if is_dev_network {
    ...
  }
  _ 
```

Reply:
ah, right, that's what I meant to do, updated